### ISSUE TITLE
Explore more complex LDAP based attacks i.e. Blind LDAP Injection

### ISSUE THREAD

**Issue by infosec-au** [digininja/vuLnDAP#2]
Hey,

Wanting to suggest an improvement for what I believe is a rarely exploited concept (Blind LDAP techniques) in application security.

The best example I can find for this is here: https://blog.ripstech.com/2017/joomla-takeover-in-20-seconds-with-ldap-injection-cve-2017-14596/

There's a handy Gif demo that demonstrates the power of these sort of attacks: 

[joomla-poc.mp4](https://blog.ripstech.com/videos/joomla-poc.mp4)

Would it be feasible for you to add a similar exercise to your project? I don't currently have the time, but if you're interested in it, I can look into it and provide input / testing / general help.

**Comment by digininja** [comment]
It doesn't seem much different to brute forcing a valid cn on this page:

<https://vulndap.digi.ninja/item?cn=robin&disp=stock,description,cn>

I'm not sure whether this type of query still works:,

```
XXX;(&(uid=Admin)(userPassword=C*))
```

From what I can tell, and I could be completely wrong here, there was a change in the way the main parsers worked which killed off a lot of the old attacks where you added an extra phrase into the query which you could then use as an "or 1=1". I didn't try with a semicolon so that might still work but from a quick test, if it did work then this should return a result and it doesn't:

<https://vulndap.digi.ninja/item?cn=robinXXX;(cn=robin)&disp=stock,description,cn>

Do you have time to find an example of how this would work using ldapsearch on the command line? If you can find a filter that you can then inject into then I should be able to implement that.

**Comment by infosec-au** [comment]
Will investigate and get back to you. I think you are correct (my bad!).

I'm currently brushing up on my LDAP bruteforcing techniques, and the first thing that came to mind was the Joomla example I showed you. 

It could be that the example I showed you is Joomla specific, or that it is no longer possible to do with the change in parsers as you've mentioned could be the case.

Nonetheless, if I come up with a way of reproducing this, I will send a PR. For now, I consider this comment closed unless someone else wants to work on this.

Thanks!

**Comment by digininja** [comment]
I'll take a PR or even a working example off some other site or the command
line. I spent quite a while banging my head trying to get injections like
the ones shown in section 2 of this working but I couldn't get them working
in any of the current implementations as they all gave syntax/parse
failures which is why I think the underlying parser was updated to prevent
extra phrases which are not syntactically correct being tagged on the end.

<https://www.owasp.org/index.php/Testing_for_LDAP_Injection_(OTG-INPVAL-006)
>

If you find ways to still do this I'd love to know as I could then
introduce more issues.
